Makefile

```diff
@@ -11,6 +11,8 @@ INSTALL = install
 INSTALL_DIR = $(INSTALL) -m755 -d
 INSTALL_DATA = $(INSTALL) -m644
 INSTALL_PROGRAM = $(INSTALL) -m755
+GIT_REVISION = $(shell (git rev-parse HEAD 2> /dev/null) || (cat .git/HEAD 2> /dev/null | head -c 40))
```

Comment:
@josevalim: you are correct about generating the wrong docs, since it will be linking to master.

Actually, this PR came initially as a one to avoid using git as a dependency, and the way I found to deal with this was to generate to files `.git_revision` and `.git_tag` that include the information there, provided git was not installed. What made me give up on this idea was that github releases doesn't allow (AFAIK) a way to execute a command when packing the release, and to generate `.git_revision` on the fly.
It's sort of pointless to make git a dependency when all we need is the revision id, but the only way solve this is a upload the release file ourselves.

One scenario that we haven't considered, is that even if git is present will not work is when someone dowloads a zip file with the source code, and tries to build the docs... there is no git information anywhere... so won't be able to build it. Therefore I think it makes sense in some way to build the release ourselves and upload, and not have git as a dependency.
what do you guys think?